### PR TITLE
Better regex for INSEE (cf #257)

### PR DESCRIPTION
**Pull request by yohanboniface**
@vinyll @christopheprudent r?

### CODE REVIEW

File: ban/core/models.py [hunk 1]
```diff
@@ -44,11 +44,12 @@ def __str__(self):
...
 
 class Municipality(NamedModel):
+    INSEE_FORMAT = '(2[AB]|\d{2})\d{3}'
     identifiers = ['siren', 'insee']
     resource_fields = ['name', 'alias', 'insee', 'siren', 'postcodes']
     exclude_for_version = ['postcodes']
 
-    insee = db.CharField(length=5, unique=True, format='[\dAB]{2}\d{3}')
+    insee = db.CharField(length=5, unique=True, format=INSEE_FORMAT)
     siren = db.CharField(length=9, format='\d*', unique=True, null=True)
```
Comment (vinyll) [inline ban/core/models.py]: I wonder if `format='\d{9}\'` would not be more explicit for error messages.

**Comment by christopheprudent** [comment]
ok w/ regex of CEA

File: ban/core/models.py [hunk 2]
```diff
@@ -119,7 +120,7 @@ def housenumbers(self):
 class HouseNumber(Model):
     # Set of characters from La Poste to optimize OCR removing confusing ones
```
Comment (vinyll) [inline ban/core/models.py]: Suggestion: _Set of OCR-friendly characters._

File: ban/core/models.py [hunk 3]
```diff
@@ -119,7 +120,7 @@ def housenumbers(self):
...
     # (like 0/O, 1/I…).
```
Comment (vinyll) [inline ban/core/models.py]: _removed confusing chars like: 0/O, 1/I…_

**Comment by yohanboniface** [comment]
@christopheprudent you should have an "appprove pull request" button somewhere then, so I can merge it :)

File: ban/core/models.py [hunk 4]
```diff
@@ -117,9 +118,9 @@ def housenumbers(self):
 
 
 class HouseNumber(Model):
-    # Set of characters from La Poste to optimize OCR removing confusing ones
-    # (like 0/O, 1/I…).
-    CEA_FORMAT = '[\dAB]{2}\d{3}[234679ABCEGHILMNPRSTUVXYZ]{5}'
+    # INSEE + set of characters from La Poste to optimize OCR (removing
```
Comment (vinyll) [inline ban/core/models.py]: Suggestion:
Set of OCR-friendly characters.
Dropped confusing chars like 0/O, 1/I, ….

File: ban/core/models.py (hunk 1 above)
Comment (yohanboniface) [inline ban/core/models.py]: I see your point.
How it is now:
- if you send a string too short or too long, you have a dedicated message
- if you send a string with the right length, but with wrong format, you'll have a format dedicated message
And we have cases with or without regex, so cases where we'd better keep the length check and dedicated message.
I feel it's good enough, but I'm open to another suggestion :)

File: ban/core/models.py (hunk 4 above)
Comment (yohanboniface) [inline ban/core/models.py]: Changed the comment with your suggestions.

**Pull request merged**